xarray/core/datatree.py

```diff
@@ -136,21 +203,25 @@ def __init__(
         raise AttributeError("DatasetView objects are not to be initialized directly")
 
     @classmethod
-    def _from_node(
+    def _from_dataset_state(
```

Comment:
I don't find this method name super intuitive - can we add an explanation comment for other devs?